Integrations/GoogleCloudStorage/GoogleCloudStorage.py

```diff
@@ -0,0 +1,566 @@
+
+    client = init_storage_client()
+    bucket = client.get_bucket(bucket_name)
+    blob = storage.Blob(blob_name, bucket)
+    saved_file_name = download_blob(blob, saved_file_name)
+
+    demisto.results(file_result_existing_file(saved_file_name))
+
+
+def gcs_upload_file():
+    entry_id = demisto.args()["entry_id"]
+    bucket_name = demisto.args()["bucket_name"]
+    object_name = demisto.args()["object_name"]
+    object_acl = demisto.args().get("object_acl", "")
+
+    context_file = demisto.getFilePath(entry_id)
+    file_path = context_file["path"]
+    file_name = context_file["name"]
+    blob = upload_blob(file_path, bucket_name, object_name)
+    if object_acl:
+        blob.acl.save_predefined(object_acl)
+
+    demisto.results({
+        "Type": entryTypes["note"],
+        "ContentsFormat": formats["text"],
+        "Contents": "File {} was successfully uploaded to bucket {} as {}".format(file_name, bucket_name, object_name)
+    })
+
+
+''' Bucket policy (ACL) '''
+
+
+def acl2dict(acl_entry, include_object_name=False):
+    """
+    Converts an ACL entry from its raw JSON form to context format (either GCS.BucketPolicy or GCS.BucketObjectPolicy).
+    """
+    result = {
+        "Bucket": acl_entry.get("bucket", "")
+    }
+
+    if include_object_name:
+        result["Object"] = acl_entry.get("object", "")
+
+    result.update({
+        "Entity": acl_entry.get("entity", ""),
+        "Email": acl_entry.get("email", ""),
+        "Role": acl_entry.get("role", ""),
+        "Team": acl_entry.get("projectTeam", {}).get("team", "")
+    })
+
+    return result
+
+
+def get_acl_entries(acl):
+    path = acl.reload_path
+    query_params = {}
+    parsed_json = acl.client._connection.api_request(method="GET", path=path, query_params=query_params)
+    return parsed_json.get("items", ())
+
+
+def set_acl_entry(acl, entity, role):
+    acl_entry = acl.entity_from_dict({"entity": entity, "role": role})
+    acl.add_entity(acl_entry)
+    acl.save()
+
+
+def delete_acl_entry(acl, entity):
+    del acl.entities[str(entity)]
+    acl.save()
+
+
+def gcs_list_bucket_policy():
+    bucket_name = demisto.args()["bucket_name"]
+
+    client = init_storage_client()
+    acl = client.get_bucket(bucket_name).acl
+
+    acl_entries = get_acl_entries(acl)
+    result = [acl2dict(entry) for entry in acl_entries]
+
+    demisto.results({
+        "Type": entryTypes["note"],
+        "ContentsFormat": formats["json"],
+        "Contents": result,
+        "HumanReadable": human_readable_table("ACL policy for bucket " + bucket_name, result),
+        "EntryContext": {ec_key("GCS.BucketPolicy", "Bucket", "Entity"): result}
+    })
+
+
+def gcs_create_bucket_policy():
+    bucket_name = demisto.args()["bucket_name"]
+    entity = demisto.args()["entity"]
+    role = demisto.args()["role"]
+
+    client = init_storage_client()
+    acl = client.get_bucket(bucket_name).acl
+    if acl.has_entity(entity):
+        return_error(
```

Comment:
it is better to raise a `ValueError` here which be caught in the `try-except` which calls `return_error`